### ISSUE TITLE
Row Grouping Support

### ISSUE THREAD

**Issue by MajedDH** [bosskmk/pluto_grid#237]
Hi,
Is there any plan to add multi level row grouping by columns support to the Grid?

Some thing like this:

![image](https://user-images.githubusercontent.com/3973115/125260525-131fe400-e309-11eb-9c9f-b62840d710e0.png)

I don't know, how hard it is to make 1-level grouping vs n-level grouping? (with/without expanders?)

If you don't mind adding the feature, or aiming to add it in the feature, maybe I'll have some time to work on it  in the near future, But of course I'll need your help getting familiar with the project.

Thanks for the great, unique package !

**Comment by bosskmk** [comment]
It's hard to add features for the time being. Thank you for your interest in the package. It would be great if many people could contribute.

**Comment by bosskmk** [comment]
I have tried many attempts to add this function in the past, but failed due to overlapping problems in the current structure.
It is difficult to group rows while maintaining the behavior of filtering, pagination, sorting, and moving cells with the keyboard.
I have tried various methods, but have not yet reached any success.
I'll try again when I have time.

The branches I tried as a prototype were 3 branches with the names feature/row-xxx.
